packages/e2e/test/k6/scenarios/wallet-restoration.test.js

```diff
@@ -54,139 +63,262 @@ const STEADY_STATE_DURATION = '5s';
+ * `wallet` MUST have at least 2 elements
+ */
+const walletDiscovery = (wallet) => {
+  check(wallet, {
+    'At least two addresses are required to run HD wallet discovery mode': (walletArray) =>
+      walletArray && walletArray.length >= 2
+  });
+
+  console.log(`Start walletDiscovery for ${wallet.length} addresses`);
+
+  // Discover stake keys derived at index > 0 on the first payment key
+  // We don't expect to find anything here so we'll use dummy addresses
+  console.debug('Discover stake keys on payment #0');
+  for (let i = 0; i < hdWalletParams.stakeAddrSearchGap; i++) {
+    const addr = getDummyAddr(wallet[0].address, i);
+    txsByAddress([addr], true, 1);
+  }
+
+  // Discover active payment addresses
+  console.debug('Discover payment addresses #1+');
+  for (const { address } of wallet) {
+    // Even if txByAddresses accepts multiple addresses, discovery does it one by one
+    txsByAddress([address], true, 1);
+  }
+
+  // Discover calls in payment address gap
+  console.debug('Discover in search gap');
+  for (let i = 0; i < hdWalletParams.paymentAddrSearchGap; i++) {
+    const addr = getDummyAddr(wallet[1].address, i);
+    txsByAddress([addr], true, 1);
+  }
+};
 
 /** Simulation of requests performed by a wallet while restoring */
 const syncWallet = ({ wallet, poolAddress }) => {
-    const startTime = Date.now();
-    const { address, stake_address: rewardAccount } = wallet;
-    cardanoHttpPost('network-info/era-summaries');
-    cardanoHttpPost('network-info/ledger-tip');
-    txsByAddress(address);
-    utxosByAddresses(address);
-    cardanoHttpPost('network-info/era-summaries');
-    cardanoHttpPost('network-info/genesis-parameters');
-    cardanoHttpPost('network-info/protocol-parameters');
-    rewardsAccBalance(rewardAccount);
-    cardanoHttpPost('network-info/ledger-tip');
-    cardanoHttpPost('network-info/lovelace-supply');
-    cardanoHttpPost('network-info/stake');
-    if (RUN_MODE === RunMode.Restore) {
-        stakePoolSearch(poolAddress);
-    }
-    cardanoHttpPost('stake-pool/stats');
+  const startTime = Date.now();
+  const addresses = wallet.map(({ address }) => address);
 
-    syncedWallets.add(wallet.address);
-    walletSyncTrend.add(Date.now() - startTime);
-    walletSyncCount.add(1);
+  if (RUN_MODE === RunMode.RestoreHD) {
+    walletDiscovery(wallet);
+  }
+
+  cardanoHttpPost('network-info/era-summaries');
+  cardanoHttpPost(TIP_URL);
+  txsByAddress(addresses);
+  utxosByAddresses(addresses);
+  cardanoHttpPost('network-info/era-summaries');
+  cardanoHttpPost('network-info/genesis-parameters');
+  cardanoHttpPost('network-info/protocol-parameters');
+  // Test restoring HD wallets with a single stake key
+  rewardsAccBalance(wallet[0].stake_address);
+  cardanoHttpPost(TIP_URL);
+  cardanoHttpPost('network-info/lovelace-supply');
+  cardanoHttpPost('network-info/stake');
+  if (RUN_MODE === RunMode.Restore) {
+    stakePoolSearch(poolAddress);
```

Comment:
Don't we need to do this for multiple pools when using multi-delegation wallets? :thinking:

Reply:
True for multi-delegation, but this test is for HD wallet (single stake key)